kong/plugins/prometheus/wasmx.lua

```diff
@@ -0,0 +1,206 @@
+local buffer = require "string.buffer"
+local wasm = require "kong.runloop.wasm"
+local wasmx_shm
+
+
+local fmt = string.format
+local str_find = string.find
+local str_match = string.match
+local str_sub = string.sub
+local table_insert = table.insert
+local table_sort = table.sort
+local buf_new = buffer.new
+local ngx_say = ngx.say
+
+
+local _M = {}
+
+
+local FLUSH_EVERY = 100
+
+
+local function sorted_iter(ctx)
+  local v = ctx.t[ctx.sorted_keys[ctx.i]]
+  ctx.i = ctx.i + 1
+
+  return v
+end
+
+
+local function sorted_pairs(t)
+  local sorted_keys = {}
+
+  for k, _ in pairs(t) do
+    table_insert(sorted_keys, k)
+  end
+
+  table_sort(sorted_keys)
+
+  return sorted_iter, { t = t, sorted_keys = sorted_keys, i = 1 }
+end
+
+
+local function parse_pw_key(key)
```

Comment:
Yeah, I intentionally avoided using regex for extracting `filter_name` and `metric_name` from the `key`.
Partially for performance, but also because I tend to avoid regex in general.

I admit, though, that the code is more readable using regex. Since we are already using regex for label extraction, I changed it to also use regex to extract `filter_name` and `metric_name`.